test/unit/core/metafunction/basic_test.cpp

```diff
@@ -36,3 +36,19 @@ TEST(metafunction, remove_cvref_)
     EXPECT_FALSE((std::is_same_v<int*, remove_cvref_t<int[3]>>));   // type stays same
     // the last example would be true for std::decay_t
 }
+
+TEST(metafunction, delete_const_t)
+{
+    EXPECT_TRUE((std::is_same_v<int, delete_const_t<int>>));
+    EXPECT_TRUE((std::is_same_v<int, delete_const_t<int const>>));
+    EXPECT_TRUE((std::is_same_v<int volatile, delete_const_t<int volatile>>));
+    EXPECT_TRUE((std::is_same_v<int &, delete_const_t<int &>>));
+    EXPECT_TRUE((std::is_same_v<int &&, delete_const_t<int &&>>));
+    EXPECT_TRUE((std::is_same_v<int &, delete_const_t<int const &>>));
+    EXPECT_TRUE((std::is_same_v<int &&, delete_const_t<int const &&>>));
+    EXPECT_TRUE((std::is_same_v<int volatile &&, delete_const_t<int const volatile &&>>));
+    EXPECT_TRUE((std::is_same_v<int *, delete_const_t<int *>>));
+    EXPECT_TRUE((std::is_same_v<int *, delete_const_t<int const *>>));
+    EXPECT_TRUE((std::is_same_v<int volatile *, delete_const_t<int const volatile *>>));
+    EXPECT_TRUE((std::is_same_v<char const *, delete_const_t<char const * const>>));
```

Comment:
Might be more readable if the two arguments are aligned vertically, but maybe that's just my taste